coral-component-table/src/scripts/Table.js

```diff
@@ -1174,6 +1180,126 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
     });
   }
 
+  /** @private */  
+  _onKeyboardDrag(event) {
+    const table = this;
+    const row = event.target.closest('tr[is="coral-table-row"]');
+    
+    if (row && table.orderable) {
+      const body = table.body;
+      const style = row.getAttribute('style');
+      const index = getIndexOf(row);
+      const oldBefore = row.nextElementSibling;
+      const dragAction = new DragAction(row);
+      const items = getRows([body]);
+
+      dragAction.axis = 'vertical';
+      // Handle the scroll in table
+      dragAction.scroll = false;
+      // Specify selection handle directly on the row if none found
+      dragAction.handle = row.querySelector('[coral-table-roworder]');
+
+      // The row placeholder indicating where the dragged element will be dropped
+      const placeholder = row.cloneNode(true);
+      placeholder.classList.add('_coral-Table-row--placeholder');
+
+      // Store the data to avoid re-reading the layout on drag events
+      const dragData = {
+        placeholder: placeholder,
+        index: index,
+        oldBefore: oldBefore,
+        // Backup styles to restore them later
+        style: {
+          row: style
+        }
+      };
+      row.dragAction._dragData = dragData;
+    }
+  }  
+
+  /** @private */
+  _onRowDragOnKeySpace(event) {
+    const dragElement = event.detail.dragElement;
+    const dragData = dragElement.dragAction._dragData;
+
+    dragData.style.cells = [];
+    getCells(dragElement).forEach((cell) => {
+      // Backup styles to restore them later
+      dragData.style.cells.push(cell.getAttribute('style'));
+      // Cells will shrink otherwise
+      cell.style.width = window.getComputedStyle(cell).width;
+    });
+  }  
+
+  /** @private */
+  _onRowDragOverOnKeyArrowDown(event) {
+    const table = this;
+    const body = table.body;
+    const dragElement = event.detail.dragElement;
+    const items = getRows([body]);
+    const index = getIndexOf(dragElement);
+    const dragData = dragElement.dragAction._dragData;
+
+    // We cannot rely on :focus since the row is being moved in the dom while dnd
+    dragElement.classList.add('is-focused');
+
+    if (dragElement === items[items.length - 1]) {
+      for (let position = 0; position < items.length - 1; position++) {
+        body.appendChild(items[position]);
+      }
+      body.insertBefore(items[0], items[items.length - 2].nextElementSibling);
+    } else {
+      body.insertBefore(items[index + 1], items[index]);
+    }
+
+    // Restore specific styling
+    dragElement.setAttribute('style', dragData.style.row || '');
+    getCells(dragElement).forEach((cell, i) => {
+      cell.setAttribute('style', dragData.style.cells[i] || '');
+    });  
```

Comment:
Restore focus to the `handle` and scroll the row into view after reordering.

Note: We should probably announce the new position of the row within the table using a live region announcement at this point.

```suggestion
    }); 
    
    handle && handle.focus();
    dragElement.scrollIntoView({block: 'nearest'});
```